geograypher/cameras/cameras.py

```diff
@@ -748,47 +763,86 @@ def save_images(self, output_folder, copy=False, remove_folder: bool = True):
     def get_lon_lat_coords(self):
         """Returns a list of GPS coords for each camera"""
         return [x.get_lon_lat() for x in self.cameras]
+    
+    def get_camera_locations(self, **kwargs):
+        """
+        Returns a list of camera locations for each camera. 
+
+        Args:
+            **kwargs: Keyword arguments to be passed to the PhotogrammetryCamera.get_camera_location method.
+
+        Returns:
+            List[Tuple[float, float] or Tuple[float, float, float]]: 
+                List of tuples containing the camera locations.
+        """
+        return [x.get_camera_location(**kwargs) for x in self.cameras]
 
     def get_subset_ROI(
         self,
-        ROI: Union[PATH_TYPE, gpd.GeoDataFrame],
-        buffer_radius_meters: float = 50,
-    ):
+        ROI: Union[PATH_TYPE, gpd.GeoDataFrame, Polygon, MultiPolygon],
+        buffer_radius: float = 0,
+        is_geospatial: bool = None
+    ):  
         """Return cameras that are within a radius of the provided geometry
 
         Args:
-            geodata (Union[PATH_TYPE, gpd.GeoDataFrame]): Geopandas dataframe or path to a geofile readable by geopandas
-            buffer_radius_meters (float, optional): Return points within this buffer of the geometry. Defaults to 50.
+            geodata (Union[PATH_TYPE, gpd.GeoDataFrame, Polygon, MultiPolygon]): 
+                This can be a Geopandas dataframe, path to a geofile readable by geopandas, or 
+                Shapely Polygon/MultiPolygon information that can be loaded into a geodataframe
+            buffer_radius (float, optional): 
+                Return points within this buffer of the geometry. Defaults to 0. Represents 
+                meters if ROI is geospatial.
+            is_geospatial (bool, optional): 
+                A flag for user to indicate if ROI is geospatial or not; if no flag is provided, 
+                the flag is set if the provided geodata has a CRS.
+        Returns:
+            subset_camera_set (List[PhotogrammetryCamera]): 
+                List of cameras that fall within the provided ROI
         """
-        if not isinstance(ROI, gpd.GeoDataFrame):
+        # construct GeoDataFrame if not provided
+        if isinstance(ROI, (Polygon,MultiPolygon)):
```

Comment:
It's odd to me that this didn't get auto-formatted to have a space after the comma. I thought that was enforced by `black`. Thoughts?

Reply:
This is strange. I ran black through the command line and it added the space after the comma. I'll look more into why this is happening.

Reply:
I think the black formatter may be running on the wrong branch because in the logs I saw that after a branch was merged black formats the code properly. I'm not sure why this is happening because `github.head_ref` represents the branch of the pull request. Since black is being triggered by isort that may be changing `github.head_ref` but this is just a guess. Is it fine if we get this branch merged after I address the rest of the comments because I would like to see if the spacing issue gets resolved after we push this to main. However, if you would not like to push unformatted code to main, I totally understand.

Reply:
I also opened an issue for this

Reply:
I'm happy to merge it as-is. Do you know if black has ever worked on a PR? I thought it had, but maybe I always saw only `isort` changes. One option would be to review the "formatting changes" commits on a couple of PRs to see what they did.